File: email-backup.java/src/main/java/demo/SessionAuthenticator.java
package demo;

import javax.mail.Authenticator;
import javax.mail.PasswordAuthentication;

final public class SessionAuthenticator extends Authenticator {

    final private String email;
    final private String password;

    public SessionAuthenticator(String email, String password) {
        this.email = email;
        this.password = password;
    }

    @Override
    public PasswordAuthentication getPasswordAuthentication() {
        return new PasswordAuthentication(email, password);
    }
}

File: email-backup.kotlin/src/main/kotlin/demo/SessionAuthenticator.kt
package demo

import javax.mail.Authenticator
import javax.mail.PasswordAuthentication

public class SessionAuthenticator(private val email: String, private val password: String) : Authenticator() {

    public override fun getPasswordAuthentication(): PasswordAuthentication {
        return PasswordAuthentication(email, password)
    }
}